src/directives/mmsTree.js

```diff
@@ -97,701 +98,173 @@ angular.module('mms.directives')
-             * @return {Object} previous branch
-             */
-            tree.get_prev_branch = function(b) {
-                var prev_sibling = tree.get_prev_sibling(b);
-                if (prev_sibling)
-                    return tree.last_descendant(prev_sibling);
-                return tree.get_parent_branch(b);
-            };
-            /**
-             * @ngdoc function
-             * @name mms.directives.directive:mmsTree#select_prev_branch
-             * @methodOf mms.directives.directive:mmsTree
-             * 
-             * @description 
-             * self explanatory
-             *
-             * @param {Object} branch current branch
-             */
-            tree.select_prev_branch = function(b) {
-                var prev = tree.get_prev_branch(b);
-                if (prev)
-                    tree.select_branch(prev);
-            };
-
-            /**
-             * @ngdoc function
-             * @name mms.directives.directive:mmsTree#refresh
-             * @methodOf mms.directives.directive:mmsTree
-             * 
-             * @description 
-             * rerender the tree when data or options change
-             */
-            tree.refresh = function() {
-                on_treeData_change();
-            };
-
-            tree.initialSelect = function() {
-                on_initialSelection_change();
-            };
-
-            tree.sort_branch = function(b, sortFunction) {
-                b.children.sort(sortFunction);
-            };
-
-            /**
-             * @ngdoc function
-             * @name mms.directives.directive:mmsTree#get_branch
-             * @methodOf mms.directives.directive:mmsTree
-             * 
-             * @description 
-             * Returns the branch with the specified data
-             */
-            tree.get_branch = function(data) {
-                var branch = null;
-                for_each_branch(function(b) {
-                    // if (angular.equals(b.data,data)) {
-                    //     branch = b;
-                    // }
-                    if (b.data.id === data.id) {
-                        branch = b;
-                    }
-                });
-                return branch;
-            };
 
-            tree.get_rows = function() {
-                return scope.tree_rows;
-            };
+    };
 
-            tree.user_clicks_branch = function(branch) {
-                return scope.user_clicks_branch(branch);
-            };
 
-        }
 
-        function getHref(row) {
-            var data = row.branch.data;
-            if (row.branch.type !== 'group' && UtilsService.isDocument(data) && !ApplicationService.getState().fullDoc) {
-                var ref = data._refId ? data._refId : 'master';
-                return UtilsService.PROJECT_URL_PREFIX + data._projectId + '/' + ref+ '/documents/' + data.id + '/views/' + data.id;
-            }
+
+    function getHref(row) {
```

Comment:
i noticed single click on a doc is now taking you to the doc instead of displaying the cover page (the double click should go to doc), it has to do with the href being populated on the anchor and the click event not being stopped somewhere, workaround is to just comment this out for now

Reply:
Yeah I noticed `project.ref.preview` seems to have just been going to the document